### PR TITLE
Implement `SearchesAdapter` for ES7.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

**Note: Requires #8482 to be merged before.**

This change is implementing the `SearchesAdapter` for ES7, creates bindings and implements the abstract integration test for it. It also includes the required classes to generate `ScrollResult` & `ScrollResult.ScrollChunk` instances.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java [hunk 1]
```diff
@@ -86,6 +90,16 @@ private SearchResponse firstResponseFrom(MultiSearchResponse result, String erro
         }
     }
 
+    public <R> R executeUnsafe(ThrowingBiFunction<RestHighLevelClient, RequestOptions, R, IOException> fn, String errorMessage) throws IOException {
```
Comment (alex-konn) [inline graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java:93]: As discussed with @dennisoelkers, this method should get a more precise name (and possibly an explanatory comment - maybe even be marked as `@Deprecated`) to indicate that it's only necessary to be compatible with existing handling of `IOException`s in scrolling-related code.

File: graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/SearchesAdapterES6.java [hunk 2]
```diff
@@ -100,7 +100,8 @@ public ScrollResult scroll(ScrollCommand scrollCommand) {
                 () -> "Unable to perform scroll search",
                 searchQuery,
                 DEFAULT_SCROLLTIME,
-                scrollCommand.fields()
+                scrollCommand.fields(),
+                scrollCommand.limit().orElse(-1)
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/SearchesAdapterES6.java:104]: :heavy_check_mark:

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchesAdapterES7.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+public class SearchesAdapterES7 implements SearchesAdapter {
+    private static final String AGG_CARDINALITY = "gl2_field_cardinality";
+    private static final String AGG_EXTENDED_STATS = "gl2_extended_stats";
+    private static final String AGG_FILTER = "gl2_filter";
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchesAdapterES7.java:33]: :heavy_check_mark:

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/Search.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.graylog.shaded.elasticsearch7.org.elasticsearch.index.query.QueryBuilders.matchAllQuery;
+import static org.graylog.shaded.elasticsearch7.org.elasticsearch.index.query.QueryBuilders.queryStringQuery;
+
+public class Search {
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/Search.java:20]: :heavy_check_mark:

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java (hunk 1 above)
Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java]: :heavy_check_mark:

File: graylog2-server/src/main/java/org/graylog2/indexer/searches/ScrollCommand.java [hunk 5]
```diff
@@ -31,6 +31,9 @@
 @AutoValue
 @JsonAutoDetect
 public abstract class ScrollCommand {
+    public static final int NO_BATCHSIZE = -1;
+    public static final int NO_LIMIT = -1;
```
Comment (alex-konn) [inline graylog2-server/src/main/java/org/graylog2/indexer/searches/ScrollCommand.java:35]: `NO_BATCHSIZE`, `NO_LIMIT`, `NO_GOD`. I like it.

**Pull request merged**